reference/performance/sql-optimizer-overview.md

```diff
@@ -18,6 +18,7 @@ category: reference
 - 谓词下推
 - 分区裁剪
 - TopN 和 Limit 下推
+- [Join Reorder](./join-reorder.md)
```

Comment:
```suggestion
- [Join Reorder](/reference/performance/join-reorder.md)
```
Links are given in a unified way in our documentation - the whole path should be written. 

Please update all the links in this PR, then the lint check will pass. @winoros